### PR TITLE
RE-1384 Make dep_update similar to standard jobs

### PR DESCRIPTION
**Pull request by git-harry**
The dep_update job is a standard job in the sense that it is designed to
be used by projects using pre/run/post scripts however the job
definition differs in a number of ways from the other standard jobs.
This change is a first pass at making the dep_update job closer in feel
to the other standard jobs.

The changes made by this commit include:
- switch the job from using long running slaves to containers.
- renaming the job files.
- renaming the jobs to have the prefix PM-Dep-Update or PR-Dep-Update.
- modifying the parameters/variables to have names consistent with the
  other standard jobs.
- the job name check in lint_jjb.py is modified to allow the first
  component to be templated.

PR test https://rpc.jenkins.cit.rackspace.net/job/PR-Dep-Update_rpc-openstack-master/2/console

PM test https://rpc.jenkins.cit.rackspace.net/job/PM-Dep-Update_rpc-openstack-master/4/console
PR generated by PM test https://github.com/rcbops/rpc-openstack/pull/2927

RE-1384 Send Jira issues to relevant projects
    
The change ensures that PM job failures are reported to the appropriate
team instead of RE given that, in general, these failures will be for
the project teams to address.

Issue: [RE-1384](https://rpc-openstack.atlassian.net/browse/RE-1384)

### CODE REVIEW

**Comment by git-harry** [comment]
recheck_all

**Comment by git-harry** [comment]
recheck_unit_tests
recheck_xenial_functonal_test

**Comment by git-harry** [comment]
recheck_xenial_functional_test

File: scripts/lint_jjb.py [hunk 1]
```diff
@@ -87,7 +87,9 @@ def parse_jjb_file(in_dir, in_file):
 
 
 def parse_job_name(job_name, file_name):
-    regex = '^([a-zA-Z0-9]+-?)+([_]([{][a-zA-Z0-9_]+[}]-?)+)?$'
+    regex = (
+        '^([a-zA-Z0-9]+-?|\{[a-zA-Z0-9]+\}-?)+([_]([{][a-zA-Z0-9_]+[}]-?)+)?$'
```
Comment (hughsaunders) [inline scripts/lint_jjb.py]: Looks ok:
image

Though the single character classes like `[_]` are a bit odd, though they are pre-existing.

File: rpc_jobs/standard_job_dep_update.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            credentials-id: "github_account_rpc_jenkins_svc"
+      script-path: job_dsl/standard_job_dep_update.groovy
+    concurrent: true
+    properties:
+      - build-discarder:
+          num-to-keep: "{NUM_TO_KEEP}"
+      - github:
+          url: "{repo_url}"
+      - inject:
+          properties-content: |
```
Comment (mattt416) [inline rpc_jobs/standard_job_dep_update.yml]: Nice!  I think I prefer this to the arbitrary env var assignment we typically do in the dsl.

Comment (git-harry) [reply rpc_jobs/standard_job_dep_update.yml]: Yes, I thought this looked a bit nicer too. Maybe worth considering applying to the other jobs at some point.

**Pull request merged**